src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,147 @@
+import { Link, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { Person } from '../../types';
+import { getSearchWith } from '../../utils/searchHelper';
+import { PersonInfo } from '../PersonInfo';
+
+interface Props {
+  people: Person[];
+  selectedPersonSlug: string;
+}
+
+enum SortType {
+  Name = 'name',
+  Sex = 'sex',
+  Born = 'born',
+  Died = 'died',
+}
+
+const compareStrings = (firstString: string, secondString: string) => {
+  return firstString.localeCompare(secondString);
+};
+
+const compareNumbers = (firstNumber: number, secondNumber: number) => {
+  return firstNumber - secondNumber;
+};
+
+const getSortedPeople = (
+  people: Person[],
+  sortType: SortType | null,
+  isReversed: boolean,
+) => {
+  let sortedPeople = [...people];
+
+  sortedPeople.sort((currentPerson, nextPerson) => {
+    switch (sortType) {
+      case SortType.Name:
+      case SortType.Sex:
+        return compareStrings(currentPerson[sortType], nextPerson[sortType]);
+
+      case SortType.Born:
+      case SortType.Died:
+        return compareNumbers(currentPerson[sortType], nextPerson[sortType]);
+
+      default: return 0;
+    }
+  });
+
+  if (isReversed) {
+    sortedPeople = sortedPeople.reverse();
+  }
+
+  return sortedPeople;
+};
+
+export const PeopleTable:React.FC<Props> = ({ people, selectedPersonSlug }) => {
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort' || '');
+  const order = searchParams.get('order' || '');
+
+  const sortedPeople = getSortedPeople(people, sort as SortType, !!order);
+
+  return (
+    <table
+      data-cy="peopleTable"
+      // eslint-disable-next-line max-len
+      className="table is-striped is-hoverable is-narrow is-fullwidth"
+    >
+
+      <thead>
+        <tr>
+          {Object.values(SortType).map(sortType => {
```

Comment:
move it to the variable
```suggestion
          {....map(sortType => {
```